airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -144,7 +144,7 @@ def _build_unload_query(
         self, credentials_block: str, select_query: str, s3_key: str, unload_options: str
     ) -> str:
         return f"""
-                    UNLOAD ('{select_query}')
+                    UNLOAD ($${select_query}$$)
```

Comment:
could you check [this comment](https://github.com/apache/airflow/pull/30300#issuecomment-1486971598)?

Reply:
Thank you for raising this concern. @vincbeck is correct that queries using double quotes might not work as expected with the current change. To address this, we can enhance the implementation to detect if the string uses double quotes, and in such cases, skip the use of `$$`. This way, we ensure compatibility for both single and double quotes.

Reply:
I am not sure this is something worth going to deep on to work around a very limited amount of edge cases. The only use of double quotes would be around column names that contain illegal character sequences. single quotes are for literals which is really going to cover 99% of usage in my opinion. @vincbeck did you have other edge cases in mind from that comment? One other way to approach this would be to simply regex replace any single quote within the string with double single quotes as given in the Redshift Unload docs and continue to wrap the entire query in single quotes. That would cover the double quote usage edge case.

Reply:
Yeah this solution can work. Or all the way around, you replace `""` with `"` and keep using `$$`. This solution MIGHT be better because, as you pointed out, queries with `""` are rare and as such the replace will be here "just in case". But I definitely think this is worth handling this use case, otherwise your change will be breaking for any query using already `""`.